db/migrate/20190630034751_create_host_facets_reported_data_facets.rb

```diff
@@ -1,7 +1,7 @@
 class CreateHostFacetsReportedDataFacets < ActiveRecord::Migration[5.2]
   def change
     create_table :host_facets_reported_data_facets do |t|
-      t.integer :host_id
+      t.references :host
```

Comment:
doesn't this imply the index?

Reply:
Darn,

```
Column `host_id` on table `host_facets_reported_data_facets` has a type of `bigint(20)`.
This does not match column `id` on `hosts`, which has type `int(11)`.
To resolve this issue, change the type of the `host_id` column on `host_facets_reported_data_facets` to be :integer. (For example `t.integer host_id`).
```

Reply:
so, reverting this

Reply:
I would rather see it fixed :( as the foreign_key/primary_key type mismatch is not a good thing to have around.

Reply:
We can pass `type: :integer` to the references to force it to be an integer.
See https://github.com/rails/rails/blob/a8dfec903bd8cd3b7df9f9d5fbb7100520049302/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb#L120

Reply:
From experience I can tell that a type mismatch destroys database performance, at least on MySQL. I've seen it ignore indexes and revert to full table scans. :+1: for fixing the root issue.